### ISSUE TITLE
Bug in HRS generation of a rule with lambdas

### ISSUE THREAD

**Issue by Rehan-MALAK** [Deducteam/lambdapi#310]
Dear all,

Here is  a ``test.lp`` with a lambda binding a dummy variable in a rule RHS :
```
constant symbol U : TYPE
constant symbol cst : U
symbol a : U ⇒ U
rule a → λdummy : U , cst
```
The generated TRS in TPDB format is (with ``lambdapi --confluence "cat > test.lp.trs" test.lp``) :
```
(FUN
  lam  : term -> (term -> term) -> term
  app  : term -> term -> term
  pi   : term -> (term -> term) -> term
  type : term
)

(COMMENT beta-reduction)
(VAR
  v_x   : term
  m_typ : term
  m_B   : term
  m_F   : term -> term
)
(RULES app(lam(m_typ,\v_x. m_F v_x), m_B) -> m_F(m_B))

(COMMENT TYPE keyword)
(FUN TYPE : term)

(COMMENT symbols)
(FUN c_test_U : term)
(FUN c_test_a : term)
(FUN c_test_cst : term)

(COMMENT rewriting rules)
(RULES c_test_a
    -> lam(c_test_U,\v_dummy.c_test_cst))
```
So \v_dummy is not declared resulting in :
```bash
$ /path/to/csiho.sh test.lp.trs
./csiho: error occured
```
Currently, in src/hrs.ml, we generate only the declaration of pattern variables in rule LHS : a lambda in a rule LHS is not working either
```
constant symbol U : TYPE
constant symbol cst : U
symbol a : (U ⇒ U) ⇒ U
rule a (λdummy,cst) → cst
```
as the generated TPDB file will also miss the declaration
```
...
(VAR
     v_dummy : term
...
```

**Comment by Rehan-MALAK** [comment]
#311

**Comment by fblanqui** [comment]
Fixed in #311
